cloudify_azure/auth/oauth2.py

```diff
@@ -76,7 +77,7 @@ def generate_jwt_token(certificate, thumbprint, client_id, talent_id):
                 talent_id),
         Jwt.EXPIRES_ON: int(time.mktime(expires.timetuple())),
         Jwt.ISSUER: client_id,
-        Jwt.JWT_ID: str(uuid.uuid4()),
+        Jwt.JWT_ID: text_type(uuid.uuid4()),
```

Comment:
That would be `unicode` in PY2.  Maybe you could use `'{0}'.format(uuid.uuid4())` instead?